src/mixer/basetrackplayer.cpp

```diff
@@ -343,22 +345,58 @@ void BaseTrackPlayerImpl::slotTrackLoaded(TrackPointer pNewTrack,
                 ConfigKey("[Mixer Profile]", "GainAutoReset"), false)) {
             m_pPreGain->set(1.0);
         }
-        int reset = m_pConfig->getValue<int>(
-                ConfigKey("[Controls]", "SpeedAutoReset"), RESET_PITCH);
-        if (reset == RESET_SPEED || reset == RESET_PITCH_AND_SPEED) {
-            // Avoid resetting speed if master sync is enabled and other decks with sync enabled
-            // are playing, as this would change the speed of already playing decks.
-            if (!m_pEngineMaster->getEngineSync()->otherSyncedPlaying(getGroup())) {
-                if (m_pRateSlider != NULL) {
-                    m_pRateSlider->set(0.0);
+
+        if (m_copyFrom.isEmpty()) {
+            int reset = m_pConfig->getValue<int>(
+                    ConfigKey("[Controls]", "SpeedAutoReset"), RESET_PITCH);
+            if (reset == RESET_SPEED || reset == RESET_PITCH_AND_SPEED) {
+                // Avoid resetting speed if master sync is enabled and other decks with sync enabled
+                // are playing, as this would change the speed of already playing decks.
+                if (!m_pEngineMaster->getEngineSync()->otherSyncedPlaying(getGroup())) {
+                    if (m_pRateSlider != NULL) {
+                        m_pRateSlider->set(0.0);
+                    }
                 }
             }
-        }
-        if (reset == RESET_PITCH || reset == RESET_PITCH_AND_SPEED) {
-            if (m_pPitchAdjust != NULL) {
-                m_pPitchAdjust->set(0.0);
+            if (reset == RESET_PITCH || reset == RESET_PITCH_AND_SPEED) {
+                if (m_pPitchAdjust != NULL) {
+                    m_pPitchAdjust->set(0.0);
+                }
+            }
+        } else {
+            ControlObject* pControl = NULL;
+
+            // copy play state
+            pControl = ControlObject::getControl(m_copyFrom, "play");
+            if (pControl) {
+                m_pPlay->set(pControl->get());
+            }
+
+            // copy rate
+            pControl = ControlObject::getControl(m_copyFrom, "rate");
+            if (pControl && m_pRateSlider != NULL) {
+                m_pRateSlider->set(pControl->get());
+            }
+
+            // copy pitch
+            pControl = ControlObject::getControl(m_copyFrom, "pitch_adjust");
+            if (pControl && m_pPitchAdjust != NULL) {
+                m_pPitchAdjust->set(pControl->get());
+            }
+
+            // copy the play position
+            pControl = ControlObject::getControl(m_copyFrom, "playposition");
+            if (pControl) {
+                m_pChannel->getEngineBuffer()->slotControlSeekExact(pControl->get());
+            }
```

Comment:
this seems to have no effect

Reply:
There is somewhere code that seeks to the cue point or 0.0 on track load.
Maybe this happens afterwards.

Reply:
I also think we have to move this code into the engine to allow an sample exact seek.
It may happen that the Playposition co is outdated. So you have to calculate the future value. You need also consider the processing order of the decks inside the engine.
There is already something similar in visualPlayposition that calculates the exact position for rendering the waveforms.

Reply:
I modified this to set `"playposition"` since the value from `"playposition"` is fractional, but the value that `slotControlSeekExact()` accepts is not. This now sets the play position correctly, but the copied track ends up slightly behind as you suspected it would.